### DISCUSSION TITLE
HSS acceleration after a gridloss

### DISCUSSION THREAD

**Discussion by Guiiem** [OpenFAST/openfast#2266] [Q&A]
Hello OpenFAST community, 

I am simulating a case with grid loss during NTM, and I am interested on the HSS acceleration (LSSGagAxa or HSShftA) right after the event.

Using the IEA 3.4MW turbine, I get this by setting `TimGenOf` to `60` and `GenTiStp` to `True` in ServoDyn. Everything in the plot is normalized to [1, 0] for comparison:

![Pasted image 20240612121016](https://github.com/OpenFAST/openfast/assets/43451847/99dae103-877c-4509-a993-985a75f321f5)

There is an artificial peak right after the generator disconnects. I have tried to reduce `DT`, but the artificial peak remains at the next timestamp after the grid-loss. Setting  `DrTrDOF` to `False` in ElastoDyn makes this peak appear also in LSSTipAxa. 

Do you think it is just a consequence of the sudden jump of generator torque, or can it be represented accurately by other means? 

Thank you!
Guillem

**Comment by jjonkman** [comment]
Dear @Guiiem,

I'm not sure.  Are you saying the spike lasts only one time regardless of the size of the time step?

Does adding a correction step in the OpenFAST glue code (`NumCrctn` > 0) eliminate the spike? Does changing the integration method in ElastoDyn (e.g,, from `Method` = 3 to 1) eliminate the spike?

Best regards,

**Reply by Guiiem** [reply]
Dear @jjonkman, 

This is exactly what I meant! And you are absolutely right, adding a correction iteration solved the issue! Setting `NumCrctn` to 1 I got this:

![image](https://github.com/OpenFAST/openfast/assets/43451847/ab950e2a-4567-441c-be7c-1ec8b4ee6837)

By setting `NumCrctn` to 2 I got exactly the same. Changing the integration method in ElastoDyn without modifying `NumCrctn` did not solve the issue. 

Thanks a lot for the quick and precise answer! 
Guillem

**Reply by jjonkman** [reply]
I'm not sure I fully understand what is causing the spike, but I'm glad adding a correction step solves the issue.

Best regards,

**Reply by bjonkman** [reply]
@jjonkman , There is some commented-out code related to variable `CheckHSSBrTrqC` in OpenFAST. When the controller does step changes, the inputs to the structural code (ElastoDyn) will need a correction step. You guys can probably generalize that `CheckHSSBrTrqC` logic for other inputs as well.

**Reply by jjonkman** [reply]
Thanks for clarifying, @bjonkman .  I've now opened issue https://github.com/OpenFAST/openfast/issues/2269 to address this.
